src/renderer/components/output.tsx

```diff
@@ -30,121 +33,195 @@ interface CommandsProps {
+          mosaicTree.splitPercentage = 25;
+        }
+        this.context.mosaicActions.replaceWith([], mosaicTree);
+      }
+    }
+  }
+
+  /**
+   * Set up Monaco Language.
+   */
+  private setupMonacoLanguage(monaco: typeof MonacoType) {
+    // register a new language
+    monaco.languages.register({ id: 'consoleOutputLanguage' });
+    // Register a tokens provider for the language
+    monaco.languages.setMonarchTokensProvider('consoleOutputLanguage', {
+      tokenizer: {
+        root: [[/\d{1,2}:\d{2}:\d{2}\s(A|P)M/, 'custom-date']],
+      },
+    });
+  }
+
+  /**
+   * Sets the model and content on the editor
+   *
+   * @private
+   * @memberof Output
+   */
+  private async setContent(output: OutputEntry[]) {
+    this.createModel(this.getOutputLines(output));
+    // have terminal always scroll to the bottom
+    this.editor.revealLine(this.editor.getScrollHeight());
+  }
+
+  /**
+   * Processes output entries such that each entry has a timestamp and value text.
+   *
    * An individual entry might span multiple lines. To ensure that
    * each line has a timestamp, this method might split up entries.
    *
    * @param {OutputEntry} entry
-   * @returns {Array<JSX.Element>}
+   * @returns string
    * @memberof Output
    */
-  public renderEntry(entry: OutputEntry, index: number): Array<JSX.Element> {
-    const ts = this.renderTimestamp(entry.timestamp);
-    const timestamp = <span className="timestamp">{ts}</span>;
-    const lines = entry.text.split(/\r?\n/);
-    const style: React.CSSProperties = entry.isNotPre
-      ? { whiteSpace: 'initial' }
-      : {};
-
-    const renderLine = (text: string, lineIndex: number): JSX.Element =>
-      isWebUri(text) ? (
-        <div key={`${entry.timestamp}--${index}--${lineIndex}`}>
-          <span style={style} className="output-message">
-            {timestamp}
-            <a onClick={() => shell.openExternal(text)}>{text}</a>
-          </span>
-        </div>
-      ) : (
-        <div key={`${entry.timestamp}--${index}--${lineIndex}`}>
-          <span style={style} className="output-message">
-            {timestamp}
-            {text}
-          </span>
-        </div>
-      );
-
-    return lines.map(renderLine);
+  private getOutputLines(output: OutputEntry[]) {
+    const { appState } = this.props;
+    const lines: string[] = [];
+    const outputs = output.slice(appState.output.length - 1000);
+
+    for (const output of outputs) {
+      const segments = output.text.split(/\r?\n/);
+      const date = new Date(output.timestamp).toLocaleTimeString();
+      for (const segment of segments) {
+        lines.push(date + ' ' + segment);
+      }
+    }
+    return lines.join('\n');
   }
 
-  public render(): JSX.Element | null {
-    const { output, isConsoleShowing } = this.props.appState;
+  /**
+   * Create a model and attach it to the editor
+   *
+   * @private
+   * @param {string} value
+   */
+  private createModel(value: string) {
```

Comment:
This function name doesn't reflect that the created model gets attached to `this.editor`, so it scans oddly to the eye when reading the code where it's called.

Maybe `setEditorText()`?

```suggestion
  private setEditorText(value: string) {
```